core/src/calleeFunctions/index.ts

```diff
@@ -67,22 +128,31 @@ export const getMarketDataById = async function (
     };
 };
 
-export const getMarketDataRecords = async function (
+const _getMarketDataRecords = async function (
     network: string,
-    collateralSymbol: string,
-    amount: BigNumber = new BigNumber('1')
+    collateralSymbol: CollateralSymbol,
+    amount: BigNumber = new BigNumber('1'),
+    useAutoRouter = false
 ): Promise<Record<string, MarketData>> {
     const collateral = getCollateralConfigBySymbol(collateralSymbol);
     let marketDataRecords = {};
     for (const marketId in collateral.exchanges) {
         let marketData: MarketData;
         try {
-            marketData = await getMarketDataById(network, collateral, marketId, amount);
+            // turn off autorouter during tests because it takes too long
+            if ((process.env.NODE_ENV === 'test' || !useAutoRouter) && marketId === 'Uniswap V3 Autorouter') {
```

Comment:
Please don't use `marketId` to determine if it's auto-routed collateral or not, you have specific key `autoRouter` for it!